### ISSUE TITLE
Wishlist: Option to view Youtube's subscribtion inbox

### ISSUE THREAD

**Issue by kuba-orlik** [mps-youtube/mps-youtube#157]
It would be trule lovely if I could browse a list of the videos by users I subscribe to - it could be done by some kind of youtube log-in functionality (but I think that's not a good solution) or by providing mps-youtube with a list uf usernames I'd like to subscribe to. This way I'd never have to load the bloaty youtube website ever again :)

**Comment by hrnr** [comment]
It would be nice to have those feature request archived somehow. I don't think it is nice to just close them, this one was for example requested multiple times.

I think it should be properly labelled (all the labels unfortunately disappeared) and kept open. Maybe someday somebody will come and implement some of those feature requests.

**Comment by ids1024** [comment]
> It would be nice to have those feature request archived somehow. I don't think it is nice to just close them, this one was for example requested multiple times.
>
> I think it should be properly labelled (all the labels unfortunately disappeared) and kept open. Maybe someday somebody will come and implement some of those feature requests.

Agreed. I think feature requests should be kept open and tagged as "feature request". With the exception of things that never will be implemented, and duplicates.

This for instance should be implemented in some form, I think. As such, you can see I added a TODO label earier. It's interesting that it shows that history even though all the labels have been lost; perhaps a script could add them back.

**Comment by NateJackDev** [comment]
Could this feature of youtube advance this feature request: https://support.google.com/youtube/answer/6224202

**Comment by bendem** [comment]
It would if there is a way to access it from the API.

**Comment by ag-eitilt** [comment]
Seeing as this has been made the central request for any type of subscription, we should probably decide what we want that to look like for when someone is able to step up and code it.  My own wishlist is rather inspired by [NewPipe](https://newpipe.schabi.org/#subscriptions), though they only provide two of the five:

* No reliance on YouTube providing the list.
    NewPipe has an explicit design decision to never log in to YouTube, and while MPS-YT doesn't need to take as strong of a stance, a lot of people are likely drawn here by the possibility of (at least partially) getting away from Google's tracking, and shouldn't have to defer right back to the company for simply searching across a pre-defined list of accounts.
* Ability to manage multiple subscriptions lists.
    A minor point, but assuming we store subscriptions locally, it should be almost trivial to allow the user to sort them by personal category.  Maybe make one list the default so people can use `subscribe <user>` if they don't want/need the power of `subscribe <user> <list>`, but provide the option if we do want to separate music from gaming from science.
* *Don't* sort subscriptions by upload.
    The main objection to the feature seems to be that YouTube doesn't provide a way of querying the subscriptions feed, which is just further hampered if we avoid requiring login.  NewPipe solves this by only loading a subset of users at a time (10 - 15 ish).  That would obviously prevent a YouTube-style interleaved subscription list, so they only show the single most recent video from each user, in what seems to be the order the queries got returned.  It might be nice to instead sort by channel name, but that sort of limited response is what I'm going for.
    Besides, doing so prevents channels which only rarely release a video from getting lost in the flood from channels which release a couple videos a day.
* Filter watched/flagged videos.
    This one is more of a long shot, but it can be nice to have the program keep track of what you've already seen in giving you a list of "new" items, rather than requiring the user remember that.  I'd like subscriptions to work with the history feature to display the *second* most recent video by a user if the first has already been watched (or the third if the second has as well, and so on).  Obviously we don't want to retrieve the entire channel history, so if we exhaust the videos in the first returned batch, we should probably just say so and not dig any deeper; probably `<N>  <Channel>: no new videos` or similar.
    Ideally this would be paired with something like #425 to be able to manually specify "I've already seen this through (an)other program(s)" or "I'm not interested in watching this video" and filter those out of the list as well.
* Allow access to a user's uploads directly from the subscription list.
    Already done!  Especially if the above filtering isn't implemented (but even after it is), I'll often want to see what some channel uploaded *between* the last video I watched and the video the list is displaying.  Just so long as we don't disable the `user` command in the features list, MPS-YT has that built in; I list it because it's what I miss the most often using NewPipe.

**Comment by yardenac** [comment]
This would be a life-changer for me.
